### PR TITLE
feat: provide dimensions for MP4

### PR DESCRIPTION
**Pull request by dominique-pfister**
Added MP4 metadata detection

### CODE REVIEW

File: src/MediaHandler.js [hunk 1]
```diff
@@ -580,11 +591,14 @@ export default class MediaHandler {
...
       if (blob.data) {
-        const { width, height } = this._getDimensions(blob.data, c);
+        const { width, height, duration } = this._getDimensions(blob.data, c);
         if (width) {
           blob.meta.width = width;
           blob.meta.height = height;
         }
+        if (duration) {
+          blob.meta.duration = duration;
+        }
```
Comment (dominique-pfister) [inline src/MediaHandler.js:601]: MP4 provides duration

File: src/MediaHandler.js [hunk 2]
```diff
@@ -743,7 +757,7 @@ export default class MediaHandler {
     } = blob;
     const ext = mime.getExtension(blob.contentType) || 'bin';
     let fragment = '';
-    if (blob.meta && blob.meta.width && blob.meta.height) {
+    if (blob.meta && blob.meta.width && blob.meta.height && !blob.contentType?.match(/^video\/[^/]+$/)) {
```
Comment (dominique-pfister) [inline src/MediaHandler.js:760]: add width and height fragment for images only

File: test/mediahandler.test.js [hunk 3]
```diff
@@ -188,6 +191,76 @@ describe('MediaHandler', () => {
...
+      hash: '122d57f2fc8e69edd784262ae09510ce81b2e0902',
+      lastModified: '01-01-2021',
+      meta: {
+        agent: `mediahandler-${version}`,
+        alg: '8k',
+        'source-last-modified': '01-01-2021',
+        src: 'https://www.example.com/test_video.mp4',
+        width: '360',
+        height: '640',
+        duration: '3',
```
Comment (dominique-pfister) [inline test/mediahandler.test.js:248]: actually a 3 seconds video from my desk

File: src/MediaHandler.js (hunk 2 above)
Comment (tripodsan) [reply src/MediaHandler.js:760]: or `&& blob.meta.duration === undefined` ?

**Review by tripodsan — APPROVED** [review]

File: src/MediaHandler.js (hunk 2 above)
Comment (dominique-pfister) [reply src/MediaHandler.js:760]: the duration **should** be available but it doesn't have to: the only information this parser is guaranteed to return is the type `video/something`.

**Pull request merged**